### PR TITLE
use `-m build` instead of particular cmds

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

simpler way of building packages
also rename action as it mainly about build not just checks

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
:books: Documentation preview :books:: https://pytorch-lightning--18635.org.readthedocs.build/en/18635/

### CODE REVIEW

File: .github/actions/pkg-build/action.yml [hunk 1]
```diff
@@ -27,7 +27,7 @@ runs:
       shell: bash
 
     - name: Create package
-      run: python setup.py sdist bdist_wheel
+      run: python -m build
```
Comment (carmocca) [inline .github/actions/pkg-build/action.yml:30]: Wbat is the advantage of -m build? The previous way is the standard and doesnt require an extra dependency

Comment (Borda) [reply .github/actions/pkg-build/action.yml:30]: `python -m build`:

- This command is part of the newer PEP 517 and PEP 518 packaging standards and is associated with the `pyproject.toml` file.
- It uses the "build backend" specified in the `pyproject.toml` file to build distribution packages.
- It is part of the modern packaging ecosystem and is recommended for new projects.
The build command is more declarative and less reliant on complex setup.py scripts.

Recommendation:

If you're starting a new Python project or working on a project that uses modern packaging standards, it's recommended to use `python -m build` along with a pyproject.toml file. This approach is more modern, standardized, and better suited for Python packaging best practices.

However, if you're working with an older project or a project that relies on `setuptools` and has a setup.py script, you should continue using `python setup.py sdist` to build your distribution package.

In summary, you can see it as a continuous dependency update...

Comment (carmocca) [reply .github/actions/pkg-build/action.yml:30]: > However, if you're working with an older project or a project that relies on setuptools and has a setup.py script, you should continue using python setup.py sdist to build your distribution package.

Don't we fall into this category? Our packaging logic is built with `setuptools` and we have a `setup.py` file

Comment (justusschock) [reply .github/actions/pkg-build/action.yml:30]: yes, but if the official recommendation is to stick to it, I agree with @carmocca to do so (especially since our `setup.py` is relatively complex)
